packages/mcfunction/src/node/macro.ts

```diff
@@ -0,0 +1,61 @@
+import * as core from '@spyglassmc/core'
+
+export interface MacroNode extends core.SequenceNode<MacroChildNode> {
+	type: 'mcfunction:macro'
+	slash?: core.Range
+	children: MacroChildNode[]
+}
+export namespace MacroNode {
+	/* istanbul ignore next */
+	export function is<T extends core.DeepReadonly<core.AstNode> | undefined>(
+		obj: T,
+	): obj is core.InheritReadonly<MacroNode, T> {
+		return (obj as MacroNode | undefined)?.type ===
+			'mcfunction:macro'
+	}
+
+	export function mock(range: core.RangeLike): MacroNode {
+		return {
+			type: 'mcfunction:macro',
+			range: core.Range.get(range),
+			children: [MacroChildNode.mock(range)],
+		}
+	}
+}
+
+export interface MacroChildNode extends core.AstNode {
+	type:
+		| 'mcfunction:macro_child/macro'
+		| 'mcfunction:macro_child/other'
```

Comment:
These two node types shouldn't share the same interface. You should have `MacroGapNode` and `MacroVariableNode` and then `MacroChildNode` can be a union of those two types.